### ISSUE TITLE
konele's text doesn't stay in input box after MIUI 14 update

### ISSUE THREAD

**Issue by nomada-rodrigo** [Kaljurand/K6nele#114]
Hi.
Since MIUI 14 update I had to remove gboard and add konele again.  But the text selected after speech recognition doesn't get inserted in the input box of whatever app I'm using. Instead it opens the box asking which app do I want to use. The app I need is never included Cuzco they are all alternative apps, I guess.

Any solutions? 

I use a redmi note 9 pro
Thanks
Rodrigo

**Comment by Kaljurand** [comment]
I have no possibility to test with a MIUI device at the moment, but maybe somebody reading this has. If possible could you add some screenshots to illustrate the issue.

**Comment by nomada-rodrigo** [comment]
Sure. Here they are.

![Screenshot_2024-01-29-10-37-34-587_org mozilla firefox](https://github.com/Kaljurand/K6nele/assets/1609557/03dadcd9-3bff-465d-b92c-5357835be02f)
![Screenshot_2024-01-29-10-38-14-393_com menny android anysoftkeyboard](https://github.com/Kaljurand/K6nele/assets/1609557/a928741e-751f-4685-9422-b74618461859)
![Screenshot_2024-01-29-10-38-24-882_ee ioc phon android speak](https://github.com/Kaljurand/K6nele/assets/1609557/0ec818d9-cd00-4daa-905f-6f94e308e335)
![Screenshot_2024-01-29-10-38-53-078_ee ioc phon android speak](https://github.com/Kaljurand/K6nele/assets/1609557/3a801352-abf6-4707-a3e5-d63f20d8159d)
![Screenshot_2024-01-29-10-39-19-798_ee ioc phon android speak](https://github.com/Kaljurand/K6nele/assets/1609557/baae2f17-8d33-4100-aad2-c1fe0af371eb)
![Screenshot_2024-01-29-10-39-28-036_android](https://github.com/Kaljurand/K6nele/assets/1609557/027c3d2d-9cd7-4cae-a8e3-094cd510783f)

**Comment by nomada-rodrigo** [comment]
Sure. Here they are.

![Screenshot_2024-01-29-10-37-34-587_org mozilla firefox](https://github.com/Kaljurand/K6nele/assets/1609557/03dadcd9-3bff-465d-b92c-5357835be02f)
![Screenshot_2024-01-29-10-38-14-393_com menny android anysoftkeyboard](https://github.com/Kaljurand/K6nele/assets/1609557/a928741e-751f-4685-9422-b74618461859)
![Screenshot_2024-01-29-10-38-24-882_ee ioc phon android speak](https://github.com/Kaljurand/K6nele/assets/1609557/0ec81
![Screenshot_2024-01-29-10-38-53-078_ee ioc phon android speak](https://github.com/Kaljurand/K6nele/assets/1609557/3a801352-abf6-4707-a3e5-d63f20d8159d)
![Screenshot_2024-01-29-10-39-19-798_ee ioc phon android speak](https://github.com/Kaljurand/K6nele/assets/1609557/baae2f17-8d33-4100-aad2-c1fe0af371eb)
![Screenshot_2024-01-29-10-39-28-036_android](https://github.com/Kaljurand/K6nele/assets/1609557/027c3d2d-9cd7-4cae-a8e3-094cd510783f)

**Comment by Kaljurand** [comment]
Looking at the screenshots, where you are using AnySoftKeyboard, it seems this issue is similar to #109 .

I've just tested with AnySoftKeyboard v1.11.177, Android v14 (build AP11.231215.009) and various speech recognizers (Kõnele, FUTO Voice, Dicio, Google, LocalSTT): inserting the recognized text into the edited text field worked with **none** of them.

It works correctly in apps like Google Photos and Google Keep, when using their own microphone button (i.e. not the one on the keyboard).

So the issue is perhaps with the particular combination of AnySoftKeyboard and the underlying OS version. I have no clear idea how to debug it further.

In general, I'd recommend to directly use Kõnele's own keyboard app (the so called "Speak & swipe keyboard") for dictating into text fields.

**Comment by nomada-rodrigo** [comment]
Hi,
When you recommend using the Könele's own keyboard, I'm not sure I understand. The app I have installed is already called "Speak and swipe keyboard" and if I activate it and deactivate ASK, I end up with only the dictate to text function (which doesn't input the result in the text field). If I click the keyboard symbol on the top left, I get a message that says "no previous keyboard".

Can you guide me on how to make that swipe keyboard available?

**Comment by nomada-rodrigo** [comment]
Exactly, the dictated text that shows when the button goes red doesn't get transferred to the text field.

I thought there was also a keyboard in "Speak and swipe", where we could type and swipe. 

I guess that means I need to find a keyboard that doesn't cause this bug with the "Speak & Swipe" app. I tried a few ones from  f-droid but no microphone button shows so that I can "turn on" the dictate to text feature of S&S, so I'm back on to gboard, but I don't like this option as the whole idea of these efforts was to abandon all these "no privacy friendly" apps.

Do you have any suggestions?

**Comment by Kaljurand** [comment]
The "Speak & swipe" offers swiping on the IME panel to move the cursor in the text field, swiping from the large microphone button to execute user-defined rewrite rules, and also typing on user-defined buttons to execute rewrite rules. But there is no traditional keyboard with the letter keys. Some examples are in this video https://www.youtube.com/watch?v=PWngf5onMaE

I am unable to reproduce the "text does not get transferred" bug. Does it only happen in certain apps (e.g. web browser), and with certain fields (e.g. HTML text fields)? Or always? I'm only aware of the issue that in the Google Docs app, when editing a file, the delete key does not work (while it works in Gboard). Screenshots (or even screencasts) would probably be helpful again.

I think Florisboard (https://github.com/florisboard/florisboard) is quite configurable and privacy friendly. You might be able to define swipes that switch to another IME (like K6nele) or configure what the microphone button does. (But I haven't tried recently.)

**Comment by nomada-rodrigo** [comment]
Thank you for your effort into this. And for the Florisboard suggestion. I'll try it out.
